### PR TITLE
Refactoring and cleanup of search engine provider controller stuffs

### PR DESCRIPTION
**Pull request by simonhong**
Issue https://github.com/brave/brave-browser/issues/1470

Please check each commit's comment.

Before migrating these stuffs to keyed service, I did some refactoring and cleanup with more test cases.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
`yarn test brave_browser_tests --filter=SearchEngineProviderControllerTest*`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/search_engine_provider_controller_browsertest.cc [hunk 1]
```diff
@@ -171,3 +172,48 @@ IN_PROC_BROWSER_TEST_F(SearchEngineProviderControllerTest,
...
+
+  // alternative pref is initially disabled.
+  EXPECT_FALSE(brave::UseAlternativeSearchEngineProviderEnabled(guest_profile));
+
+  brave::ToggleUseAlternativeSearchEngineProvider(guest_profile);
+  EXPECT_TRUE(brave::UseAlternativeSearchEngineProviderEnabled(guest_profile));
+  int provider_id =
+      TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO;
+
+  //Check tor profile's search provider is set to ddg.
```
Comment (darkdh) [inline browser/search_engine_provider_controller_browsertest.cc]: s/tor/guest/

File: browser/guest_window_search_engine_provider_controller.h [hunk 2]
```diff
@@ -7,6 +7,11 @@
 
 #include "brave/browser/search_engine_provider_controller_base.h"
 
+// This controller is only used by non Qwant region.
+// This class manage's newtab's ddg toggle button state.
+// Toggle button state should be reflected setting value.
+// Ex, when user changeds from ddg to others, toggle button shouldbe off and
```
Comment (darkdh) [inline browser/guest_window_search_engine_provider_controller.h]: s/changeds/changes/
s/shouldbe/should be/

File: browser/search_engine_provider_controller_browsertest.cc (hunk 1 above)
Comment (simonhong) [reply browser/search_engine_provider_controller_browsertest.cc]: Done.

File: browser/guest_window_search_engine_provider_controller.h (hunk 2 above)
Comment (simonhong) [reply browser/guest_window_search_engine_provider_controller.h]: Done. thanks for review!

**Pull request merged**